src/definitions.jl

```diff
@@ -693,40 +680,57 @@ adjoint_mul(p::Plan, x::AbstractArray) = adjoint_mul(p, x, AdjointStyle(p))
 function adjoint_mul(p::Plan{T}, x::AbstractArray, ::FFTAdjointStyle) where {T}
     dims = fftdims(p)
     N = normalization(T, size(p), dims)
-    return (p \ x) / N
+    pinv = inv(p)
+    # Optimization: when pinv is a ScaledPlan, check if we can avoid a loop over x.
+    # Even if not, ensure that we do only one pass by combining the normalization with the plan.
+    if pinv isa ScaledPlan && pinv.scale == N
+        return pinv.p * x
+    else
+        return (1/N * pinv) * x
+    end
 end
 
 function adjoint_mul(p::Plan{T}, x::AbstractArray, ::RFFTAdjointStyle) where {T<:Real}
     dims = fftdims(p)
     N = normalization(T, size(p), dims)
     halfdim = first(dims)
     d = size(p, halfdim)
-    n = output_size(p, halfdim)
+    pinv = inv(p)
+    n = size(pinv, halfdim)
+    # Optimization: when pinv is a ScaledPlan, fuse the scaling into our map to ensure we do not loop over x twice.
+    scale = pinv isa ScaledPlan ? pinv.scale / 2N : 1 / 2N
```

Comment:
Same here, this introduces `Float64` promotions.

Reply:
```julia
julia> using FFTW

julia> P = plan_rfft(rand(Float32, 3, 3));

julia> P' * (P * rand(Float32, 3, 3))
3×3 Matrix{Float32}:
 1.35893  3.17063  3.99985
 1.15958  6.03116  3.67565
 3.10326  5.80633  3.89977
```